### DISCUSSION TITLE
How do I reference ninja in vcpkg

### DISCUSSION THREAD

**Discussion by ThomsonTan** [microsoft/vcpkg#19139] [Ideas]
vcpkg includes a binary download of ninja.exe for its `vcpkg install`, is it possible to reference this ninja.exe in a project which depend on vcpkg (`cmake -G Ninja -DCMAKE_TOOLCHAIN_FILE=.../vcpkg.cmake`), without hard-code the path of ninja.exe with version number?

**Comment by linquize** [comment]
Use shell script, to choose the latest ninja version in `downloads/tools` directory
```sh
dir=$(find downloads/tools -name 'ninja-*' | sort --version-sort | tail -n1)
```
add ninja directory to path
```sh
export PATH=$dir:$PATH
```

**Comment by ThomsonTan** [comment]
@linquize search ninja executable looks slightly better than hard-code the path, but still doesn't looks like an idea solution. As vcpkg already downloads cmake and ninja, it will be great if there is a documented way to the projects using vcpkg to access them, like vcpkg export their path bases on some command line option. This could help CI build environment by saving the extra config of both tools.

**Comment by JackBoosY** [comment]
We need to consider compatibility issues caused by different versions.
